### PR TITLE
Fix TriggeredDagRunOperator triggered link

### PR DESCRIPTION
**Pull request by fredthomsen**
Link to the specific dag run id that was triggered, as opposed to the logical date, such that the instance that was triggered on the DAG grid UI.

This was alluded to in #19264, but you want the DAG run that was triggered select in the DAG grid UI.

Current link and DAG grid UI view:
Screenshot 2024-06-19 at 19 18 08

This PR's changed link and DAG grid UI view:
Screenshot 2024-06-19 at 19 38 17

Simple DAGs to demonstrate:
```
from airflow.decorators import dag, task
from airflow.operators.trigger_dagrun import TriggerDagRunOperator


@dag()
def triggering_dag():
    t = TriggerDagRunOperator(
        trigger_dag_id="triggered_dag",
        task_id="triggering_task",
    )

    t

@dag()
def triggered_dag():
    @task()
    def empty_task() -> int:
        return 0

    empty_task()

triggering_dag()
triggered_dag()
```

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/operators/trigger_dagrun.py [hunk 1]
```diff
@@ -218,7 +221,7 @@ def execute(self, context: Context):
                 raise e
         if dag_run is None:
             raise RuntimeError("The dag_run should be set here!")
-        # Store the execution date from the dag run (either created or found above) to
+        # Store the run id from the dag run (either created or found above) to
         # be used when creating the extra link on the webserver.
```
Comment (uranusjr) [inline airflow/operators/trigger_dagrun.py:225]: We should add a todo note here that execution date is stored only for backward compatibility and should be removed in Airflow 3.

Comment (fredthomsen) [reply airflow/operators/trigger_dagrun.py:225]: Ok.  Update pushed.

**Review by uranusjr — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Comment by sbrinkhorst** [comment]
Hi all, 

I think the URL should include both the execution_date and the dag_run_id. Because if you only include the dag_run_id it will not work if that value is not in the 25 latest dag runs (or whatever you've set as `default_dag_run_display_number`).

**Comment by eladkal** [comment]
> Hi all,
> 
> I think the URL should include both the execution_date and the dag_run_id. Because if you only include the dag_run_id it will not work if that value is not in the 25 latest dag runs (or whatever you've set as `default_dag_run_display_number`).

Why not?

**Comment by eladkal** [comment]
If you refer to https://github.com/apache/airflow/issues/39642 it has already been fixed

**Comment by sbrinkhorst** [comment]
Has the way url's are processed changed since the release of 2.9.2? Because now I see changes in how the url's are generated, but not in how they are processed. Because I'm unable to build the latest version locally, so I'm testing on 2.9.2, but I can manually create url's based on what I see in the code/screenshots.

If I look at the url visible in the screenshot and substitute the values for my own setup, I get an URL that doesn't work for DagRuns not in the latest 25. So I have `https://<url>/dags/<dagName>/grid?dag_run_id=<runId>` for an older run it doesn't work. It takes the execution date of the dag run I refer to, but only shows runs older than that date. So the right most run is the one before the run I'm actually looking for.

If I make it `https://<url>/dags/<dagName>/grid?dag_run_id=<runId>&execution_date=<date>` the correct run is actually visible and selected.

(I'm quite sceptical about #396442 actually being fixed by that MR.)

**Comment by eladkal** [comment]
The PR I shared is targeted to 2.9.3 which hasnt been released yet.

If you want to report a problem please first check with main branch. If you find a bug you can open Github issue with the reporduce example

**Pull request merged**